Sources/Fuzzilli/FuzzIL/TypeSystem.swift

```diff
@@ -850,53 +789,168 @@ class TypeExtension: Hashable {
     }
 }
 
+public enum Parameter: Equatable, Hashable {
+    case plain(Type)
+    case opt(Type)
+    case rest(Type)
+
+    fileprivate func format(abbreviate: Bool) -> String {
+        switch self {
+            case .plain(let t):
+                return ".plain(\(t.format(abbreviate: abbreviate)))"
+            case .opt(let t):
+                return ".opt(\(t.format(abbreviate: abbreviate)))"
+            case .rest(let t):
+               return ".rest(\(t.format(abbreviate: abbreviate)))"
+        }
+    }
+
+    fileprivate func callerType() -> Type {
```

Comment:
Nit: I think we could turn these two methods into getters (`var callerType: Type { ...`) to save a few keystrokes when using them, since they are essentially trivial getters.